api_service_instances.go

```diff
@@ -1,888 +1,753 @@
-		if localVarHTTPResponse.StatusCode == 401 {
-			var v Error
-			err = a.client.decode(&v, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
-				return localVarReturnValue, localVarHTTPResponse, newErr
-			}
-			newErr.model = v
-			return localVarReturnValue, localVarHTTPResponse, newErr
-		}
-		if localVarHTTPResponse.StatusCode == 410 {
-			var v Error
-			err = a.client.decode(&v, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
-				return localVarReturnValue, localVarHTTPResponse, newErr
-			}
-			newErr.model = v
-			return localVarReturnValue, localVarHTTPResponse, newErr
-		}
-		if localVarHTTPResponse.StatusCode == 422 {
-			var v Error
-			err = a.client.decode(&v, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
-				return localVarReturnValue, localVarHTTPResponse, newErr
-			}
-			newErr.model = v
-		}
-		return localVarReturnValue, localVarHTTPResponse, newErr
-	}
-
-	err = a.client.decode(&localVarReturnValue, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
+	localVarBody, err := ioutil.ReadAll(localVarHttpResponse.Body)
+	localVarHttpResponse.Body.Close()
 	if err != nil {
-		newErr := GenericOpenAPIError{
-			body:  localVarBody,
-			error: err.Error(),
-		}
-		return localVarReturnValue, localVarHTTPResponse, newErr
+		return localVarReturnValue, localVarHttpResponse, err
 	}
 
-	return localVarReturnValue, localVarHTTPResponse, nil
-}
-
-type ApiServiceInstanceGetRequest struct {
-	ctx _context.Context
-	ApiService *ServiceInstancesApiService
-	xBrokerAPIVersion *string
-	instanceId string
-	xBrokerAPIOriginatingIdentity *string
-	serviceId *string
-	planId *string
-}
+	if localVarHttpResponse.StatusCode < 300 {
+		// If we succeed, return the data, otherwise pass on to decode error.
+		err = a.client.decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"));
+		if err == nil { 
+			return localVarReturnValue, localVarHttpResponse, err
+		}
+	}
 
-func (r ApiServiceInstanceGetRequest) XBrokerAPIVersion(xBrokerAPIVersion string) ApiServiceInstanceGetRequest {
-	r.xBrokerAPIVersion = &xBrokerAPIVersion
-	return r
-}
-func (r ApiServiceInstanceGetRequest) XBrokerAPIOriginatingIdentity(xBrokerAPIOriginatingIdentity string) ApiServiceInstanceGetRequest {
-	r.xBrokerAPIOriginatingIdentity = &xBrokerAPIOriginatingIdentity
-	return r
-}
-func (r ApiServiceInstanceGetRequest) ServiceId(serviceId string) ApiServiceInstanceGetRequest {
-	r.serviceId = &serviceId
-	return r
-}
-func (r ApiServiceInstanceGetRequest) PlanId(planId string) ApiServiceInstanceGetRequest {
-	r.planId = &planId
-	return r
-}
+	if localVarHttpResponse.StatusCode >= 300 {
+		newErr := GenericSwaggerError{
+			body: localVarBody,
+			error: localVarHttpResponse.Status,
+		}
+		if localVarHttpResponse.StatusCode == 200 {
```

Comment:
This version is differentiating the 2xx codes(!) but I think the logic is unreachable. Specifically: 

```
if status >= 300 {
    if status == 200 {
    }
}
```

Reply:
I think this will run into the same issue. 

```
if localVarHttpResponse.StatusCode <= 300 {
    err = a.client.decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"));	 
    if err == nil {
        return localVarReturnValue, localVarHttpResponse, err
    }
}
```

Where `localVarReturnValue` is a `ServiceInstanceProvisionResponse` (synchronous response)

Reply:
Fortunately, the unreachable 202 conditions properly returns the `ServiceInstanceAsyncOperation` -- a step in the right direction.